transactionsx/constants.go

```diff
@@ -55,26 +55,25 @@ func (state TransactionAttemptState) String() string {
 }
 
 // TransactionErrorReason is the reason why a transaction should be failed.
-// Internal: This should never be used and is not supported.
 type TransactionErrorReason uint8
 
 // NOTE: The errors within this section are critically ordered, as the order of
 // precedence used when merging errors together is based on this.
 const (
 	// TransactionErrorReasonSuccess indicates the transaction succeeded and did not fail.
-	TransactionErrorReasonSuccess TransactionErrorReason = iota
+	TransactionErrorReasonSuccess TransactionErrorReason = TransactionErrorReason(0)
 
 	// TransactionErrorReasonTransactionFailed indicates the transaction should be failed because it failed.
-	TransactionErrorReasonTransactionFailed
+	TransactionErrorReasonTransactionFailed = TransactionErrorReason(1)
```

Comment:
Should all these not have the Transaction prefix?

Reply:
I left the Transaction prefix on things that are specific to the Transaction object.